web/client-api/src/main/java/io/deephaven/web/client/state/ClientTableState.java

```diff
@@ -57,6 +57,81 @@
  * Consider making this a js type with restricted, read-only property access.
  */
 public final class ClientTableState extends TableConfig {
+    public ChunkType[] chunkTypes() {
+        // This is roughly ReinterpretUtils.maybeConvertToWritablePrimitiveChunkType, and should
+        // be rewritten to skip the trip through Class
+        return Arrays.stream(columnTypes()).map(dataType -> {
+            if (dataType == Boolean.class || dataType == boolean.class) {
+                return ChunkType.Object;
+            }
+            // JS client holds date objects as objects, not as longs
+            // if (dataType == DateWrapper.class) {
```

Comment:
TODO: either finish/fix/remove or add a GH issue

Reply:
Sorry, this was meant to serve as a comment about how it works, not a suggestion of how to fix, will simplify/clarify.

Reply:
Also will move these into the main part of the class, rather than above the constructor/fields ... not sure why IJ refactors them here.